app-android/src/main/java/com/ustadmobile/port/android/view/Login2Fragment.kt

```diff
@@ -119,5 +142,94 @@ class Login2Fragment : UstadBaseFragment(), Login2View {
         mPresenter = null
         mBinding = null
     }
+}
+
+@Composable
+private fun LoginScreen(
+    uiState: LoginUiState = LoginUiState(),
+    onClickLogin: () -> Unit = {},
+    onClickCreateAccount: () -> Unit = {},
+    onClickConnectAsGuest: () -> Unit = {},
+    onUsernameValueChange: (String) -> Unit = {},
+    onPasswordValueChange: (String) -> Unit = {},
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally
+    )  {
+
+        UstadTextEditField(
+            value = uiState.username,
+            label = stringResource(id = R.string.username),
+            onValueChange = {
+                onUsernameValueChange(it)
+            },
+            error = uiState.usernameError,
+            enabled = uiState.usernameEnabled,
+        )
+
+        UstadTextEditField(
+            value = uiState.password,
+            label = stringResource(id = R.string.password),
+            onValueChange = {
+                onPasswordValueChange(it)
+            },
+            error = uiState.passwordError,
+            enabled = uiState.passwordEnabled,
+            trailingIcon = {
+                Icon(
+                    painter = painterResource(R.drawable.ic_baseline_visibility_24),
+                    contentDescription = "")
+            }
+        )
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        Button(
+            onClick = onClickLogin,
+            modifier = Modifier
+                .fillMaxWidth(),
+            colors = ButtonDefaults.buttonColors(
+                contentColor = Color.Black,
```

Comment:
We should never use hardcoded colors! What if the secondary color is also black? If there is a particular background, use contentColorFor